src/ui/public/vislib/visualizations/pie_chart.js

```diff
@@ -162,6 +184,126 @@ export function VislibVisualizationsPieChartProvider(Private) {
         return color(d.name);
       });
 
+      // add labels
+      if (showLabels) {
+        const labelGroups = labels
+          .datum(slices)
+          .selectAll('.label')
+          .data(partition.nodes);
+
+        // create an empty quadtree to hold label positions
+        const svgBBox = {
+          width: svg.node().parentElement.clientWidth,
+          height: svg.node().parentElement.clientHeight
+        };
+
+        const labelLayout = d3.geom.quadtree()
+          .extent([[-svgBBox.width,-svgBBox.height], [svgBBox.width, svgBBox.height] ])
+          .x(function (d) { return d.position.x; })
+          .y(function (d) { return d.position.y; })
+          ([]);
+
+        labelGroups
+          .enter()
+          .append('g')
+          .attr('class', 'label')
+          .append('text')
+          .text(function (d) {
+            if (d.depth === 0) {
+              return;
```

Comment:
If you add `this.parentElement.remove()` here, that wouldn't leave an empty `g` element for the actual inner donut circle.